tests/unit/ingestion/models/ingestionManager.spec.ts

```diff
@@ -455,4 +481,426 @@ describe('IngestionManager', () => {
+          },
+        },
+      };
+      const mockValidationTask = {
+        id: taskId,
+        jobId,
+        type: 'validation',
+        status: OperationStatus.COMPLETED,
+        parameters: {
+          isValid: true,
+          checksums: [],
+        },
+      };
+
+      getJobSpy.mockResolvedValue(mockJob);
+      getTasksForJobSpy.mockResolvedValue([mockValidationTask]);
+      resetJobSpy.mockResolvedValue(undefined);
+      zodValidator.validate.mockResolvedValue(undefined);
+      mockPolygonPartsManagerClient.deleteValidationEntity.mockResolvedValue(undefined);
+
+      await ingestionManager.retryIngestion(jobId);
+
+      expect(resetJobSpy).toHaveBeenCalledWith(jobId);
+    });
+
+    it('should update task with new checksums when shapefile has changed and job is SUSPENDED', async () => {
+      const jobId = faker.string.uuid();
+      const taskId = faker.string.uuid();
+      const oldChecksum = 'oldChecksum123';
+      const newChecksum = 'newChecksum456';
+
+      const mockJob = {
+        id: jobId,
+        status: OperationStatus.SUSPENDED,
+        productType: 'Orthophoto',
+        resourceId: rasterLayerMetadataGenerators.productId(),
+        parameters: {
+          inputFiles: {
+            gpkgFilesPath: ['/path/to/file.gpkg'],
+            metadataShapefilePath: 'metadata.shp',
+            productShapefilePath: '/path/to/product.shp',
+          },
+        },
+      };
+      const mockValidationTask = {
+        id: taskId,
+        jobId,
+        type: 'validation',
+        status: OperationStatus.COMPLETED,
+        parameters: {
+          isValid: false,
+          checksums: [{ fileName: 'metadata.shp', checksum: oldChecksum }],
+        },
+      };
+
+      getJobSpy.mockResolvedValue(mockJob);
+      getTasksForJobSpy.mockResolvedValue([mockValidationTask]);
+      // Mock calculateChecksum to return unique filenames based on the input path
+      calcualteChecksumSpy.mockImplementation(async (filePath: string) =>
+        Promise.resolve({
+          fileName: filePath,
+          checksum: newChecksum,
+        })
+      );
+      updateTaskSpy.mockResolvedValue(undefined);
+      updateJobSpy.mockResolvedValue(undefined);
+      zodValidator.validate.mockResolvedValue(undefined);
+      sourceValidator.validateFilesExist.mockResolvedValue(undefined);
+      mockPolygonPartsManagerClient.deleteValidationEntity.mockResolvedValue(undefined);
+
+      await ingestionManager.retryIngestion(jobId);
+
+      expect(updateTaskSpy).toHaveBeenCalledTimes(1);
+
+      const callArgs = updateTaskSpy.mock.calls[0] as [
+        string,
+        string,
+        { status: string; attempts: number; parameters: { isValid: boolean; checksums: unknown[] } }
+      ];
+      const [calledJobId, calledTaskId, calledParams] = callArgs;
+
+      expect(calledJobId).toBe(jobId);
+      expect(calledTaskId).toBe(taskId);
+      expect(calledParams).toEqual(
```

Comment:
is that `calledParams` are job or task param? please mention it to be more clearly